From 644ed719668385bd4cfe09f38372d722f6547e92 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:28:42 -0400
Subject: [acpi] libata: Store an ACPI device pointer in struct acpi_dev_node

Message-id: <1411129776-21656-8-git-send-email-prarit@redhat.com>
Patchwork-id: 96343
O-Subject: [RHEL7.1 PATCH BZ 1141846 07/61] ACPI / driver core: Store an ACPI device pointer in struct acpi_dev_node
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

Backport the missing libata-acpi.c chunk of

commit 7b1998116bbb2f3e5dd6cb9a8ee6db479b0b50a9
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Mon Nov 11 22:41:56 2013 +0100

    ACPI / driver core: Store an ACPI device pointer in struct acpi_dev_node

The remaining code was brought into RHEL7 in commit
5e4bfe0a822167bd9a7f62ed8026ced9b8afadc3 which contains the note:

      ./drivers/ata/libata-acpi.c
        Hunks #1 and #2 is not applicable to RHEL7 due to upstream commit
        f1bc1e4 - ata: acpi: rework the ata acpi bind support - not having
        been brought into RHEL7.

This is, of course, no longer the case with this patchset.

Cc: David Milburn <dmilburn@redhat.com>
Cc: Jeff Garzik <jgarzik@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/ata/libata-acpi.c b/drivers/ata/libata-acpi.c
index ab714d2..4372cfa 100644
--- a/drivers/ata/libata-acpi.c
+++ b/drivers/ata/libata-acpi.c
@@ -185,7 +185,7 @@ void ata_acpi_bind_port(struct ata_port *ap)
  if (libata_noacpi || ap->flags & ATA_FLAG_ACPI_SATA || !host_handle)
   return;
 
- ACPI_HANDLE_SET(&ap->tdev, acpi_get_child(host_handle, ap->port_no));
+ acpi_preset_companion(&ap->tdev, host_handle, ap->port_no);
 
  if (ata_acpi_gtm(ap, &ap->__acpi_init_gtm) == 0)
   ap->pflags |= ATA_PFLAG_INIT_GTM_VALID;
@@ -222,7 +222,7 @@ void ata_acpi_bind_dev(struct ata_device *dev)
   parent_handle = port_handle;
  }
 
- ACPI_HANDLE_SET(&dev->tdev, acpi_get_child(parent_handle, adr));
+ acpi_preset_companion(&dev->tdev, parent_handle, adr);
 
  register_hotplug_dock_device(ata_dev_acpi_handle(dev),
          &ata_acpi_dev_dock_ops, dev, NULL, NULL);
-- 
1.7.1